src/command.ts

```diff
@@ -0,0 +1,261 @@
+    });
+
+    if (this.helpText) {
+      cmd.on("--help", () => {
+        console.log(this.helpText);
+      });
+    }
+
+    cmd.action((...args: any[]) => {
+      const runner = this.runner();
+      const start = new Date().getTime();
+      const options = last(args);
+      const argCount = cmd._args.length;
+      if (args.length - 1 > argCount) {
+        client.errorOut(
+          new FirebaseError(
+            `Too many arguments. Run ${bold("firebase help " + cmd._name)} for usage instructions`,
+            { exit: 1 }
+          )
+        );
+        return;
+      }
+
+      runner(...args)
+        .then((result) => {
+          if (getInheritedOption(options, "json")) {
+            console.log(
+              JSON.stringify(
+                {
+                  status: "success",
+                  result: result,
+                },
+                null,
+                2
+              )
+            );
+          }
+          const duration = new Date().getTime() - start;
+          track(this.name, "success", duration).then(process.exit);
+        })
+        .catch(async (err) => {
+          if (getInheritedOption(options, "json")) {
+            console.log(
+              JSON.stringify(
+                {
+                  status: "error",
+                  error: err.message,
+                },
+                null,
+                2
+              )
+            );
+          }
+          const duration = Date.now() - start;
+          const errorEvent = err.exit === 1 ? "Error (User)" : "Error (Unexpected)";
+
+          await Promise.all([track(this.name, "error", duration), track(errorEvent, "", duration)]);
+          client.errorOut(err);
+        });
+    });
+  }
+
+  /**
+   * Extends the options with various properties for use in commands.
+   * @param options the command options object.
+   */
+  private prepare(options: any): void {
+    options = options || {};
+    options.project = getInheritedOption(options, "project");
+
+    if (!process.stdin.isTTY || getInheritedOption(options, "nonInteractive")) {
+      options.nonInteractive = true;
+    }
+    // allow override of detected non-interactive with --interactive flag
+    if (getInheritedOption(options, "interactive")) {
+      options.nonInteractive = false;
+    }
+
+    if (getInheritedOption(options, "debug")) {
+      logger.transports.console.level = "debug";
+      options.debug = true;
+    }
+    if (getInheritedOption(options, "json")) {
+      options.nonInteractive = true;
+      logger.transports.console.level = "none";
+    }
+
+    try {
+      options.config = _load(options);
+    } catch (e) {
+      options.configError = e;
+    }
+
+    options.projectRoot = detectProjectRoot(options.cwd);
+    this.applyRC(options);
+  }
+
+  /**
+   * Apply configuration from .firebaserc files in the working directory tree.
+   * @param options the command options object.
+   */
+  applyRC(options: any): void {
```

Comment:
Should this be `private`?  Should any other methods here?

Reply:
it should, good call - I thought I caught them all, but I didn't...